### DISCUSSION TITLE
Distributed sensor not getting its config

### DISCUSSION THREAD

**Discussion by rosswakelin** [Security-Onion-Solutions/securityonion#11093] [2.4] [answered]
I've just installed my first 2.4.10 sensor on dedicated hardware, and I a a bit confused on what happens.
The sensor install is VERY quick, with no evidence of any docker containers being installed. 
The sensor pops up in the "pending" list on the Grid, and when accepted - nothing happens.  The sensor  disappears from the pending list, then pops up again there a few minutes later.  Here is a bit of the salt master log from the manager:
```
2023-08-18 02:53:02,354 [salt.channel.server:288 ][INFO    ][2025] Authentication request from chl-osoids-p01_sensor
2023-08-18 02:53:02,354 [salt.channel.server:564 ][INFO    ][2025] Authentication accepted from chl-osoids-p01_sensor
2023-08-18 02:53:03,940 [salt.client.mixins:440 ][INFO    ][2022] Runner completed: 20230818025303878136
2023-08-18 02:53:04,702 [salt.client.mixins:440 ][INFO    ][2022] Runner completed: 20230818025304267075
2023-08-18 02:53:04,720 [salt.pillar      :893 ][ERROR   ][2022] Specified SLS 'minions.chl-osoids-p01_sensor' in environment 'base' is not available on the salt master
2023-08-18 02:53:04,721 [salt.template    :62  ][ERROR   ][2022] Template was specified incorrectly: False
2023-08-18 02:53:04,721 [salt.pillar      :893 ][ERROR   ][2022] Specified SLS 'minions.adv_chl-osoids-p01_sensor' in environment 'base' is not available on the salt master
2023-08-18 02:53:04,721 [salt.template    :62  ][ERROR   ][2022] Template was specified incorrectly: False
2023-08-18 02:53:04,722 [salt.pillar      :1256][CRITICAL][2022] Pillar render error: Specified SLS 'minions.chl-osoids-p01_sensor' in environment 'base' is not available on the salt master
2023-08-18 02:53:04,722 [salt.pillar      :1256][CRITICAL][2022] Pillar render error: Specified SLS 'minions.adv_chl-osoids-p01_sensor' in environment 'base' is not available on the salt master
2023-08-18 02:53:04,969 [salt.utils.job   :86  ][INFO    ][2019] Got return from chl-osoids-p01_sensor for job 20230818025304969051
2023-08-18 02:53:04,970 [salt.utils.event :911 ][ERROR   ][2019] Event iteration failed with exception: 'list' object has no attribute 'items'

```
The above happens when you accept the sensor, then stops when the sensor goes back to pending.
There are no firewalls between the sensor and the manager and the manager has the sensor subnet as allowed in the hostgroups.

Any hints?

**Comment by Pencat89** [comment]
On the manager, go into Administration, Grid Members, then accept them

**Comment by rosswakelin** [comment]
I already did that. As I said above, it reappears in the pending list.Sent from my Galaxy
-------- Original message --------From: Pencat89 ***@***.***> Date: 18/08/23  18:32  (GMT+12:00) To: Security-Onion-Solutions/securityonion ***@***.***> Cc: rosswakelin ***@***.***>, Author ***@***.***> Subject: Re: [Security-Onion-Solutions/securityonion] Distributed sensor not getting its config (Discussion #11093) 
On the manager, go into Administration, Grid Members, then accept them

**Comment by dougburks** [comment]
Please provide more information as shown at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Please include the following details where you can:
Security Onion version as seen in the lower left corner of SOC and in /etc/soversion. For example: 2.4.10
Is this a cloud deployment or on-prem? If on-prem, do you have Internet access or this an airgap installation?
Did you install from our Security Onion ISO image or did you perform a network installation?
If network installation, what distro and version did you install on?
How many nodes do you have?
What are the hardware specs of each of those nodes?
How are each of those nodes configured? (ex. manager with 2 search nodes and 3 forward nodes)
Are you experiencing issues monitoring network traffic? If so, are you sniffing from a tap or span port and what is the traffic volume?
Does so-status show all services running?
Do you get any failures when you run sudo salt-call state.highstate?
Does the SOC Grid page show any failures?

**Comment by rosswakelin** [comment]
More details:
SO 2.4.10
On-prem, internet access
Installed from ISO
One manager node, two search nodes, one receiver node (not working, separate issue), one sensor (this one, faulty)
Manager/search/receiver are virtual, more than adequate resources.
Sensor is physical, output from so-setup here:
```
2023-08-18T02:19:35Z | INFO | Executing command: lscpu
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   46 bits physical, 48 bits virtual
Byte Order:                      Little Endian
CPU(s):                          8
On-line CPU(s) list:             0-7
Vendor ID:                       GenuineIntel
BIOS Vendor ID:                  Intel
Model name:                      Intel(R) Xeon(R) CPU D-1518 @ 2.20GHz
BIOS Model name:                 Intel(R) Xeon(R) CPU D-1518 @ 2.20GHz
CPU family:                      6
Model:                           86
Thread(s) per core:              2
Core(s) per socket:              4
Socket(s):                       1
Stepping:                        3
CPU max MHz:                     2200.0000
CPU min MHz:                     800.0000
BogoMIPS:                        4399.77
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopolog
y nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb
cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap intel_pt xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm arat pln pts
 md_clear flush_l1d
L1d cache:                       128 KiB (4 instances)
L1i cache:                       128 KiB (4 instances)
L2 cache:                        1 MiB (4 instances)
L3 cache:                        6 MiB (1 instance)
NUMA node(s):                    1
NUMA node0 CPU(s):               0-7
Vulnerability Itlb multihit:     KVM: Mitigation: VMX unsupported
Vulnerability L1tf:              Mitigation; PTE Inversion
Vulnerability Mds:               Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Mmio stale data:   Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Mitigation; Clear CPU buffers; SMT vulnerable
```
Sniffing is via switch span ports.
Hardware was running SO 2.3.260 just fine.
so-status responds with so-status not yet available.

Here is the tail of sosetup.log:
```
2023-08-18T02:27:59Z | INFO | Executing command: salt-call state.apply setup.highstate_cron --local --file-root=../salt/
[INFO    ] Loading fresh modules for state activity
[INFO    ] Fetching file from saltenv 'base', ** done ** 'setup/highstate_cron.sls'
[INFO    ] Running state [PATH=$PATH:/usr/sbin salt-call state.highstate] at time 02:28:00.969372
[INFO    ] Executing state cron.present for [PATH=$PATH:/usr/sbin salt-call state.highstate]
[INFO    ] Executing command 'crontab' in directory '/root'
[INFO    ] Executing command 'crontab' in directory '/root'
[INFO    ] {'root': 'PATH=$PATH:/usr/sbin salt-call state.highstate'}
[INFO    ] Completed state [PATH=$PATH:/usr/sbin salt-call state.highstate] at time 02:28:00.987029 (duration_in_ms=17.657)
local:
----------
          ID: post_setup_cron
    Function: cron.present
        Name: PATH=$PATH:/usr/sbin salt-call state.highstate
      Result: True
     Comment: Cron PATH=$PATH:/usr/sbin salt-call state.highstate added to root's crontab
     Started: 02:28:00.969372
    Duration: 17.657 ms
     Changes:
              ----------
              root:
                  PATH=$PATH:/usr/sbin salt-call state.highstate

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  17.657 ms
2023-08-18T02:28:01Z | INFO | Verifying setup
Successfully completed setup!
```
When I accept the sensor in the grid, I get this error after about 20 seconds:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/22168385/6961cbb3-4b09-4817-8021-d822c836e919)
and the sensor stays in the "Pending members" area.

**Comment by rosswakelin** [comment]
Given that I'm now seeing three different errors on this new install that indicate some files are missing from the install, I'm going to blow away my grid and start again.....

**Comment by defensivedepth** [comment, marked as answer]
There is currently a known issue with Receiver nodes, it will be fixed in an upcoming hotfix.

**Reply by 8bitjoe** [reply]
and sensor nodes?

**Reply by defensivedepth** [reply]
@8bitjoe There is not a known issue with sensor nodes. I would suggest opening up a new Discussion and we can troubleshoot from there.

**Reply by rosswakelin** [reply]
As I said above, I was seeing so many issues I blew the whole grid away, and rebuilt using HF1 and had no issues.
